tests/bluetooth/audio/bap_broadcast_source/src/main.c

```diff
@@ -506,32 +514,100 @@ ZTEST_F(bap_broadcast_source_test_suite, test_broadcast_source_start_inval_doubl
 	zassert_not_equal(0, err, "Did not fail with starting already started source");
 }
 
-ZTEST_F(bap_broadcast_source_test_suite, test_broadcast_source_reconfigure)
+ZTEST_F(bap_broadcast_source_test_suite, test_broadcast_source_reconfigure_single_subgroup)
 {
 	struct bt_bap_broadcast_source_param *reconf_param = fixture->param;
-	struct bt_bap_stream *stream = reconf_param->params[0].params[0].stream;
+	const size_t subgroup_cnt = reconf_param->params_count;
 	int err;
 
 	printk("Creating broadcast source with %zu subgroups with %zu streams\n",
 	       reconf_param->params_count, fixture->stream_cnt);
 
 	err = bt_bap_broadcast_source_create(reconf_param, &fixture->source);
+	zassert_equal(0, err, "Unable to create broadcast source: err %d", err);
```

Comment:
I'm just wondering if we aren't making test case `test_broadcast_source_create_delete` redundant here. Same goes in the next test case also.

I'm just wondering, if "create" fails, is it better that we are not testing (zassert) that functionality in other functions also, so to get a cleaner error log?

Reply:
I really dont think `test_broadcast_source_create_delete` should be removed, its a very clear test case imo. I was more leaning to whether we actually need to perform the assert after performing create in the other test cases. But as I think of it, if we dont, those will output irrelevant errors if it was "create" that failed. So I think it should stay as is also.